opm/simulators/flow_ebos_oilwater_polymer.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "config.h"
+
+// Define making clear that the simulator supports AMG
+#define FLOW_SUPPORT_AMG 1
+
+#include <opm/simulators/flow_ebos_oilwater_polymer.hpp>
+
+#include <opm/material/common/ResetLocale.hpp>
+#include <ewoms/models/blackoil/blackoiltwophaseindices.hh>
+
+#include <opm/grid/CpGrid.hpp>
+#include <opm/autodiff/SimulatorFullyImplicitBlackoilEbos.hpp>
+#include <opm/autodiff/FlowMainEbos.hpp>
+
+#if HAVE_DUNE_FEM
+#include <dune/fem/misc/mpimanager.hh>
+#else
+#include <dune/common/parallel/mpihelper.hh>
+#endif
+
+namespace Ewoms {
+namespace Properties {
+NEW_TYPE_TAG(EclFlowOilWaterPolymerProblem, INHERITS_FROM(EclFlowProblem));
+SET_BOOL_PROP(EclFlowOilWaterPolymerProblem, EnablePolymer, true);
+//! The indices required by the model
+SET_TYPE_PROP(EclFlowOilWaterPolymerProblem, Indices,
+              Ewoms::BlackOilTwoPhaseIndices<GET_PROP_VALUE(TypeTag, EnableSolvent)?1:0,
+                GET_PROP_VALUE(TypeTag, EnablePolymer)?1:0,
+                /*PVOffset=*/0, /*disabledCompIdx=*/2>);
```

Comment:
Try to change 
Polymer  + 2p is never tested. I guess I found a bug. 
Try this patch.  
```
/home/thsa/workspace/opm/ewoms/ewoms/models/blackoil/blackoiltwophaseindices.hh
-   static const int polymerConcentrationIdx  = solventSaturationIdx + numPolymers;
+  static const int polymerConcentrationIdx  = solventSaturationIdx + numSolvents;

```

Reply:
Thanks. I am creating a fix in ewoms side.

Reply:
My suggestion was something like 
```
-    static const int solventSaturationIdx  = PVOffset + numPhases;
+    static const int solventSaturationIdx  = numSolvents > 0 ? PVOffset + numPhases : -10000;
 
     //! Index of the primary variable for the first polymer
-    static const int polymerConcentrationIdx  = solventSaturationIdx + numPolymers;
+    static const int polymerConcentrationIdx  = numPolymers > 0 ?
+                (numSolvents > 0? solventSaturationIdx + numPolymers : PVOffset + numPhases) : -10000;
```

Reply:
>
```
/home/thsa/workspace/opm/ewoms/ewoms/models/blackoil/blackoiltwophaseindices.hh
-   static const int polymerConcentrationIdx  = solventSaturationIdx + numPolymers;
+  static const int polymerConcentrationIdx  = solventSaturationIdx + numSolvents;
```

I will take your suggestion. It is simpler.